### PR TITLE
proxy/backendtest: fix flaky TestServerShouldCloseWhenAllRequestsAreFulfilled

### LINKED ISSUE #2723: Flaky TestServerShouldCloseWhenAllRequestsAreFulfilled
```
--- FAIL: TestServerShouldCloseWhenAllRequestsAreFulfilled (0.02s)
    backendrecorder_test.go:28: number of requests served does not match with the expected. Expected 4 but got 0
    backendrecorder_test.go:18: Get "http://127.0.0.1:46517/2": read tcp 127.0.0.1:54354->127.0.0.1:46517: read: connection reset by peer
FAIL
FAIL	github.com/zalando/skipper/proxy/backendtest	0.065s
```

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Wait for test requests to complete before closing the recorder server via Done().

Fixes #2723

### CODE REVIEW

File: proxy/backendtest/backendrecorder_test.go [hunk 1]
```diff
@@ -3,28 +3,33 @@ package backendtest
...
 func TestServerShouldCloseWhenAllRequestsAreFulfilled(t *testing.T) {
 	expectedRequests := 4
 	recorder := NewBackendRecorder(10 * time.Millisecond)
+
+	var g errgroup.Group
 	for i := 0; i < expectedRequests; i++ {
-		go func(counter int) {
-			resp, err := http.Get(recorder.GetURL() + "/" + strconv.Itoa(counter))
+		g.Go(func() error {
+			resp, err := http.Get(recorder.GetURL())
```
Comment (RomanZavodskikh) [inline proxy/backendtest/backendrecorder_test.go:21]: Is it okay to change URL in this case?

**Comment by szuecs** [comment]
:+1:

File: proxy/backendtest/backendrecorder_test.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply proxy/backendtest/backendrecorder_test.go:21]: yeah, it does not matter

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by RomanZavodskikh** [comment]
:+1:

**Pull request merged**